crates/client/data-availability/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+pub mod avail;
+pub mod celestia;
+pub mod ethereum;
+mod sharp;
+pub mod utils;
+
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use ethers::types::{I256, U256};
+use futures::StreamExt;
+use sc_client_api::client::BlockchainEvents;
+use serde::Deserialize;
+use sp_api::ProvideRuntimeApi;
+use sp_runtime::traits::Block as BlockT;
+
+pub type StorageWrites<'a> = Vec<(&'a [u8], &'a [u8])>;
+
+pub struct DataAvailabilityWorker<B, C>(PhantomData<(B, C)>);
+
+#[derive(Debug, Copy, Clone, PartialEq, clap::ValueEnum)]
+pub enum DaLayer {
+    Celestia,
+    Ethereum,
+    Avail,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Deserialize, Default)]
+pub enum DaMode {
+    #[serde(rename = "validity")]
+    Validity,
+    #[serde(rename = "volition")]
+    Volition,
+    #[serde(rename = "validium")]
+    #[default]
+    Validium,
+}
```

Comment:
Can we have a lil' bit of doc on what those modes imply?

Reply:
doc added